### ISSUE TITLE
Empires summary window does not account for Engineering Corps cost

### ISSUE THREAD

**Issue by Blue-Award** [freeorion/freeorion#4740]
Bug Report
==========

This is on recent builds but probably goes way back. I haven't used Engineering Corps policy in past, but did so in slow MP25 game, so you'll have saves for that (adopted since turn 179)

Basically what the title says. The IP cost on the upper bar correctly shows the expected Influence Points gain for the next turn. However, Empires window summary that shows known start-of-turn values does not account for IP cost of all those ships paying influence points for Engineering Corps

I would expect the Empires summary shows start-of-turn IP gain value including Engineering Corps policy costs
![IP view](https://github.com/freeorion/freeorion/assets/110989598/b59c0b87-159c-4714-b9a8-192e1efae797)

**Comment by Blue-Award** [comment]
Not sure what to do for sensor ghosts of enemy players. Actually does the ship count on the Empires window include ghosts you didn't dismiss? Then probably it's only fair to include them also in IP costs

**Comment by geoffthemedio** [comment]
I don't think the tallies for the empires list knows about ghosts or dismissing them.

Otherwise, #4764 might fix it?

**Comment by Blue-Award** [comment]
In MP25 turn 226 (present turn as of writing), without these changes I see 4 IP for myself in the Empires window. With this change, i see -8.3. But I have 92 ships, so it should be -9.2 difference, no? Dunno where do extra 3 IP diff come from. 

There are also some slight science point generation differences for some reason

Upper is with this change applied, lower is without. But you know, server doesn't have those changes so... dunno

![Resource_gen_tally_226](https://github.com/freeorion/freeorion/assets/110989598/c3f6bad8-a1d0-4fa7-8d72-e359c0540099)

**Comment by Blue-Award** [comment]
OK so I've added the `!this_client_known_destroyed_objects.contains(obj.ID()) && !this_client_stale_object_info.contains(obj.ID());` condition to the mix so those aren't counted (quick and dirty added to `owned_by_empire`) and it looks better - no changes in science (probably some stale/destroyed data from ships in blizzards?) and IP also more like expected I think? though it's still 8.6 rather than 9.2 but I've noticed monster ships have Influence meter of 0.0 even though Target Influence is -0.1, and I have 6 of them, so it matches that I don't get that extra -0.6 penalty (so, a new bug/issue perhaps, why those monster ships don't change my influence?)

![Resource_gen_tally_226](https://github.com/freeorion/freeorion/assets/110989598/f4ecada2-e901-4717-bb2b-c74d75908ef9)

**Comment by Blue-Award** [comment]
so I don't know what those stale data are but it's probably that rather than destroyed, cause I can see destroyed on object list and they don't have science or influence attached. Could prolly attach debugger and confirm it comes from the stale.. or delete just the stale part of condition

**Comment by geoffthemedio** [comment]
An option would be to just show the current values of a few empire statistics as calculated on the server for the graphs in the pedia.

**Comment by Blue-Award** [comment]
graphs I think leak total information and are routinely disabled in multiplayer games for that reason, meanwhile "Empires" window is calculated based on what you know, which sounds better (just needs to take stock of this "stale" data issue)

**Comment by geoffthemedio** [comment]
Yes, they are different. Using the statistics is still an option, though.
